File: LeetCode/美团/station.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int shortestDistance(int n, vector<int> a, int x, int y) {
    vector<int> dp(n, 0);
    dp[x - 1] = a[x - 1];
    for (int i = x; i < y; ++i) {
        dp[i] = min(dp[i - 1] + a[i - 1], dp[x - 1] + a[i - 1]);
    }
    return dp[y - 1];
}

int main() {
    int n = 5;
    vector<int> a = {2, 3, 4, 5, 1};
    int x = 1;
    int y = 4;
    
    int result = shortestDistance(n, a, x, y);
    cout << "最短距离：" << result << endl;

    return 0;
}

File: LeetCode/美团/station.py
def shortestDistance(n:int, a:list[int], x:int,y:int):
    dp = [0] *n
    dp[x-1] = 0
    result1=0
    result2=0
    for i in range(x-1,y-1):
        result1+=a[i]
    print(result1)
    for j in range(y-1,x-1+n):
        result2+=a[j]
    return min(result1,result2)
# 示例输入
n = 5
a =[2,3,4,5,1]
x=1
y=4
#输出最短距离
print(shortestDistance(n,a,x,y))
